cpu/stm32_common/periph/i2c_2.c

```diff
@@ -79,9 +79,10 @@ void i2c_init(i2c_t dev)
 
     mutex_init(&locks[dev]);
 
+#if DEVELHELP
     I2C_TypeDef *i2c = i2c_config[dev].dev;
-
     assert(i2c != NULL);
```

Comment:
I think if we can avoid ifdefs that would be nice,
would you consider just doing 
```c
assert(i2c_config[dev].dev != NULL);
```

Reply:
agree! will push your suggestion